taker-frontend/src/components/CloseButton.tsx

```diff
@@ -0,0 +1,129 @@
+
+    const disableCloseButton = cfd.state.getGroup() === StateGroupKey.CLOSED
+        || !(cfd.state.key === StateKey.OPEN);
+
+    let closeButton;
+
+    if (connectedToMaker) {
+        closeButton = <Box w={"45%"}>
+            <Popover
+                placement="bottom"
+                closeOnBlur={true}
+            >
+                {({ onClose }) => (<>
+                    <PopoverTrigger>
+                        <Button colorScheme={"blue"} disabled={disableCloseButton}>Close</Button>
+                    </PopoverTrigger>
+                    <PopoverContent color="white" bg="blue.800" borderColor="blue.800">
+                        <PopoverHeader pt={4} fontWeight="bold" border="0">
+                            Collaboratively Close your position
+                        </PopoverHeader>
+                        <PopoverArrow />
+                        <PopoverCloseButton />
+                        <PopoverBody>
+                            <Text>
+                                This will collaboritively close your position with the counterparty. The exchange rate
+                                at {cfd.expiry_timestamp}
+                                will determine your profit/losses. It is likely that the rate will change until then.
+                            </Text>
+                        </PopoverBody>
+                        <PopoverFooter
+                            border="0"
+                            d="flex"
+                            alignItems="center"
+                            justifyContent="space-between"
+                            pb={4}
+                        >
+                            <Button
+                                size="sm"
+                                colorScheme="red"
+                                onClick={() => {
+                                    console.log(`Settling CFD ${cfd.order_id}`);
+                                    settle({});
+                                    onClose();
+                                }}
+                                isLoading={isSettling}
+                            >
+                                Close
+                            </Button>
+                        </PopoverFooter>
+                    </PopoverContent>
+                </>)}
+            </Popover>
+        </Box>;
+    } else {
+        closeButton = <Box w={"45%"}>
+            <Popover
+                placement="bottom"
+                closeOnBlur={true}
+            >
+                {({ onClose }) => (<>
+                    <PopoverTrigger>
+                        <Button colorScheme={"blue"} disabled={disableCloseButton}>Commit</Button>
+                    </PopoverTrigger>
+                    <PopoverContent color="white" bg="blue.800" borderColor="blue.800">
+                        <PopoverHeader pt={4} fontWeight="bold" border="0">
+                            Non-Collaboratively Close your position
```

Comment:
@bonomat sometimes calls this "force-close".

Thoughts? Don't need to discuss this at length now, the text can easily be changed later :)

Reply:
I prefer force-close but non-collaboratively close works as well.. no hard feelings here :)